vm/src/obj/objstr.rs

```diff
@@ -967,22 +1020,55 @@ fn adjust_indices(
     }
 }
 
-// helper function to title strings
-fn make_title(s: &str) -> String {
-    let mut titled_str = String::new();
-    let mut capitalize_char: bool = true;
-    for c in s.chars() {
-        if c.is_alphabetic() {
-            if !capitalize_char {
-                titled_str.push(c);
-            } else if capitalize_char {
-                titled_str.push(c.to_ascii_uppercase());
-                capitalize_char = false;
-            }
-        } else {
-            titled_str.push(c);
-            capitalize_char = true;
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn str_title() {
+        let vm = VirtualMachine::new();
+
+        let tests = vec![
+            (" Hello ", " hello "),
+            ("Hello ", "hello "),
+            ("Hello ", "Hello "),
+            ("Format This As Title String", "fOrMaT thIs aS titLe String"),
+            ("Format,This-As*Title;String", "fOrMaT,thIs-aS*titLe;String"),
+            ("Getint", "getInt"),
+            ("Greek Ωppercases ...", "greek ωppercases ..."),
```

Comment:
Nit: you don't need to allocate using `String::from`, you can just directly compare `title` and `&PyString::...`. I think.